tests/unit_tests/sparse_blas/source/sparse_spmv_buffer.cpp

```diff
@@ -0,0 +1,236 @@
+        shuffle_sparse_matrix(format, indexing, ia_host.data(), ja_host.data(), a_host.data(), nnz,
+                              static_cast<std::size_t>(nrows_A));
+    }
+
+    auto ia_buf = make_buffer(ia_host);
+    auto ja_buf = make_buffer(ja_host);
+    auto a_buf = make_buffer(a_host);
+    auto x_buf = make_buffer(x_host);
+    auto y_buf = make_buffer(y_host);
+
+    oneapi::mkl::sparse::matrix_handle_t A_handle = nullptr;
+    oneapi::mkl::sparse::dense_vector_handle_t x_handle = nullptr;
+    oneapi::mkl::sparse::dense_vector_handle_t y_handle = nullptr;
+    oneapi::mkl::sparse::spmv_descr_t descr = nullptr;
+    try {
+        init_sparse_matrix(main_queue, format, &A_handle, nrows_A, ncols_A, nnz, index, ia_buf,
+                           ja_buf, a_buf);
+        for (auto property : matrix_properties) {
+            CALL_RT_OR_CT(oneapi::mkl::sparse::set_matrix_property, main_queue, A_handle, property);
+        }
+        CALL_RT_OR_CT(oneapi::mkl::sparse::init_dense_vector, main_queue, &x_handle,
+                      static_cast<std::int64_t>(x_host.size()), x_buf);
+        CALL_RT_OR_CT(oneapi::mkl::sparse::init_dense_vector, main_queue, &y_handle,
+                      static_cast<std::int64_t>(y_host.size()), y_buf);
+
+        CALL_RT_OR_CT(oneapi::mkl::sparse::init_spmv_descr, main_queue, &descr);
+
+        std::size_t workspace_size = 0;
+        CALL_RT_OR_CT(oneapi::mkl::sparse::spmv_buffer_size, main_queue, transpose_val, &alpha,
+                      A_view, A_handle, x_handle, &beta, y_handle, alg, descr, workspace_size);
+        sycl::buffer<std::uint8_t, 1> workspace_buf((sycl::range<1>(workspace_size)));
+
+        CALL_RT_OR_CT(oneapi::mkl::sparse::spmv_optimize, main_queue, transpose_val, &alpha, A_view,
+                      A_handle, x_handle, &beta, y_handle, alg, descr, workspace_buf);
+
+        CALL_RT_OR_CT(oneapi::mkl::sparse::spmv, main_queue, transpose_val, &alpha, A_view,
+                      A_handle, x_handle, &beta, y_handle, alg, descr);
+
+        if (reset_data) {
+            intType reset_nnz = generate_random_matrix<fpType, intType>(
+                format, nrows_A, ncols_A, density_A_matrix, indexing, ia_host, ja_host, a_host,
+                is_symmetric);
+            if (!is_sorted) {
+                shuffle_sparse_matrix(format, indexing, ia_host.data(), ja_host.data(),
+                                      a_host.data(), reset_nnz, static_cast<std::size_t>(nrows_A));
+            }
+            if (reset_nnz > nnz) {
+                ia_buf = make_buffer(ia_host);
+                ja_buf = make_buffer(ja_host);
+                a_buf = make_buffer(a_host);
+            }
+            else {
+                copy_host_to_buffer(main_queue, ia_host, ia_buf);
+                copy_host_to_buffer(main_queue, ja_host, ja_buf);
+                copy_host_to_buffer(main_queue, a_host, a_buf);
+            }
+            fill_buffer_to_0(main_queue, y_buf);
```

Comment:
I have changed this in https://github.com/oneapi-src/oneMKL/pull/500/commits/0c34cfcf7c98982733126dd6b79dacd74b61f9dc so these buffers are initialized with random values like the first test.
This piece of code is only hit when we test re-using the same data handles (`reset_data=true`) which happens to have `beta=0`. The new version ensures that this would also work for any value of `beta`.

Reply:
Great!